shared/interfaces/CredentialInterface.ts

```diff
@@ -0,0 +1,4 @@
+export interface Credentials {
```

Comment:
shared folder sounds a nice idea, but it will create extra hustle for us with the ts config to include this folder. It is fine to repeat those interface between front and back at this point.